packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx

```diff
@@ -196,5 +196,29 @@ describe('<DataGrid /> - Toolbar', () => {
 
       expect(getColumnHeadersTextContent()).to.deep.equal(['id', 'brand']);
     });
+
+    it('should keep the focus on the switch after toggling a column', () => {
+      const { getByRole } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            components={{
+              Toolbar: GridToolbar,
+            }}
+          />
+        </div>,
+      );
+
+      const button = getByRole('button', { name: 'Select columns' });
```

Comment:
After hesitations with Sebastian, we have been leaning toward the global approach, simpler

```suggestion
      const button = screen.getByRole('button', { name: 'Select columns' });
```